core/src/main/scala/kafka/utils/NetworkClientBlockingOps.scala

```diff
@@ -56,8 +56,13 @@ class NetworkClientBlockingOps(val client: NetworkClient) extends AnyVal {
    */
   def blockingReady(node: Node, timeout: Long)(implicit time: JTime): Boolean = {
     require(timeout >=0, "timeout should be >= 0")
-    client.ready(node, time.milliseconds()) || pollUntil(timeout) { (_, now) =>
-      if (client.isReady(node, now))
+
+    // poll once to receive pending disconnects
+    val now = time.milliseconds()
+    client.poll(0, now)
+
+    client.ready(node, now) || pollUntil(timeout) { (_, now) =>
+      if (client.ready(node, now))
```

Comment:
I think the change is good for the reason you said. I was just making sure I understood the reason. :)